extensions/amp-video/0.1/flexible-bitrate.js

```diff
@@ -285,8 +285,8 @@ export class BitrateManager {
  */
 function onNontrivialWait(video, callback) {
   listen(video, 'waiting', () => {
-    // Do not trigger downgrade if not loaded metadata yet.
-    if (video.readyState < 1) {
+    // Do not trigger downgrade if not loaded metadata yet, or if video is fully loaded (eg: replay).
+    if (video.readyState < 1 || getBufferedPercentage(video) === 1) {
```

Comment:
Just making sure you tested cross browser that it does add up to 1, I'm kinda scare of additions in JavaScript :))

Reply:
The additions on `getBufferedPercentage` are rarely executed since we load videos linearly now, so there's only one buffer in the video. Multiple buffers can be found if you seek the video to an unloaded section, so a new buffer starts loading from that timestamp. However, for peace of mind we can use `> 0.99` in case there's multiple buffers in the future